src/io/flutter/sdk/FlutterSdkChannel.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.sdk;
+
+import org.jetbrains.annotations.NotNull;
+
+public class FlutterSdkChannel {
+
+  public enum ID {
+
+    // Do not change this order.
+    STABLE("stable"), BETA("beta"), DEV("dev"), MASTER("master");
+
+    private final String name;
+
+    ID(String name) {
+      this.name = name;
+    }
+
+    public String toString() {
+      return name;
+    }
+
+    @NotNull
+    public static ID fromText(String name) {
+      switch (name) {
+        case "master":
+          return MASTER;
+        case "dev":
+          return DEV;
+        case "beta":
+          return BETA;
+        case "stable":
+          return STABLE;
+        default:
+          throw new IllegalArgumentException("Unknown channel: " + name);
```

Comment:
Do we want to throw here? Return null?

For a longer tail of users, the channel text will be just the git branch they're on for the flutter repo. We may want to handle unknown names, and map them to something like 'unknown' (instead of it being an error condition).